sympy/core/numbers.py

```diff
@@ -477,15 +477,15 @@ def mod_inverse(a, m):
     - https://en.wikipedia.org/wiki/Modular_multiplicative_inverse
     - https://en.wikipedia.org/wiki/Extended_Euclidean_algorithm
     """
-    c = None
+    c = m
+    def inv(a, m):
+        if -2 < a < 2: return a
+        return m + (1 - m * inv(m % a, a)) // a
```

Comment:
The current version of mod_inverse, based on igcdex, is non-recursive. This makes it recursive. What are the implications for very large a and m (running time, stack overflow)?

Reply:
Running Time is same. (Recursive call is similar to that used in computation of gcd).
Mainly, there is no extraneous variables required for gcd and the coefficients.
Though, there might be a slight slowdown due to stack overhead.
I will stress test for performance and post the results.

Reply:
@normalhuman Tested for 2 random 8192-bit coprime numbers. Turns out that apart from having stack overhead, the recursion limit gets exceeded quickly. 
So, switched back to extended euclidean algorithm. Thanks for the feedback.